sql/pre_install/tables.sql

```diff
@@ -207,6 +207,46 @@ CREATE INDEX chunk_index_hypertable_id_hypertable_index_name_idx ON _timescaledb
 
 SELECT pg_catalog.pg_extension_config_dump('_timescaledb_catalog.chunk_index', '');
 
+-- A dimension range defines a range for a given column in a chunk. It
+-- tracks the min and the max value for that column in the specific chunk.
+--
+-- A "special" entry for a column with invalid chunk_id, PG_INT64_MAX,
+-- PG_INT64_MIN indicates that ranges could be computed for this column
+-- for chunks.
+--
+-- The ranges can overlap across chunks. The values could be out-of-date if
+-- modifications/changes occur in the corresponding chunk and such entries
+-- should be marked as "invalid" to ensure that the chunk is in
+-- appropriate state to be able to use these values. Thus these entries
+-- are different from dimension_slice which is used for tracking partitioning
+-- column ranges which have different characteristics.
+--
+-- Currently this catalog supports datatypes like INT, SERIAL, BIGSERIAL,
+-- DATE, TIMESTAMP etc. by storing the ranges in bigint columns. In the
+-- future, we could support additional datatypes (which support btree style
+-- >, <, = comparators) by storing their textual representation.
+--
+CREATE TABLE _timescaledb_catalog.dimension_range (
+  id serial NOT NULL,
+  hypertable_id integer NOT NULL,
```

Comment:
Do we strictly need the `hypertable_id` here? I guess it is fine to denormalize if for a good reason, but not clear.

I am also wondering if we really need the serial ID primary key. Could we just use `<chunk_id, column_name>`?

Actually, I am not even sure we need a primary key as I am not sure anything will reference this column. A unique constraint on <chunk_id, column_name> should be enough.

Reply:
@erimatnor when we add a dimension range, we add an entry in this catalog which is of the form

`ht_id, INVALID_chunk_id, column_name, -INF, +INF`. How do we differentiate between two same named columns from two different hypertables that need these dimension ranges otherwise? 

In terms of the serial ID primary key, I think that's just good DB practices. Future use cases might need to directly refer an entry in this catalog and the serial ID could come handy there.